### ISSUE TITLE
Feature: RTI verify installed via CLI

### ISSUE THREAD

**Issue by elgeeko1** [lf-lang/lingua-franca#2191]
User story: As a Lingua Franca developer and user, I would like to know that the C RTI is installed and executable, such as a `--version` or `--is-present` command that exits with return code 0 if the RTI binary is installed and runnable, so that I can verify development environments I provision.

Background: Docker, Ansible, Multipass, VMWare, CodePods, etc., are used to build and distribute development environments. `lfc` and `lingo` both support a `--version` command that prints the version of the application and exits with return code 0. This allows a baseline verification that the tools have been installed and can load without missing dependencies. The C RTI, however, has no such argument, and instead only `grep`-ing the output of `RTI` with no arguments can be used to infer the binary is installed and runnable. This solution is brittle.

Instead, either a `--version`, `--status`, `--is-present` or similar would allow for a primitive verification that it has been compiled and installed correctly.

**Comment by erlingrj** [comment]
Agreed. RTI should support `rti --version` where it prints out version, commit-id and date, just like `lfc`. We haven't discussed how we should version the RTI yet. I think the most elegant way of achieving a `--version` option is to have CMake, at compile-time, check the git history and find the most recent tag (which should correspond to a release) and use this as the version see this blog: https://embeddedartistry.com/blog/2016/12/21/giving-your-firmware-build-a-version/

**Comment by lhstrh** [comment]
Related conversation https://github.com/lf-lang/reactor-c/pull/348#discussion_r1482532664

**Comment by elgeeko1** [comment]
The latest tag may not always correspond with the latest release (a developer can push any arbitrary tag), but I did find out how to get the latest release tag for downloading and installing lingo:

```shell
export lingo_version=$(git ls-remote --refs --sort="version:refname" --tags https://github.com/lf-lang/lingo | cut -d/ -f3- | tail -n1)
git clone --single-branch --depth 1 --branch ${lingo_version} https://github.com/lf-lang/lingo
cargo install --path ./lingo
rm -rf /var/cache/lf-lang
```

**Comment by lhstrh** [comment]
It's a nice idea to use tags to determine what the build version is. That's less work than maintaining a version number manually, but it will not allow us to distinguish between some random (nightly/snapshot) build and the actual latest release...

**Comment by elgeeko1** [comment]
The `git ls-remote` command above returns a sorted list of all tags (`--tags`) associated with a release (`--sort="version:refname") which should return the tag associated with the latest release.

**Comment by lhstrh** [comment]
Yes, that makes sense, but by the time there are commits on the branch _since_ that that, there is no way of knowing that it's actually not the release version but something else. We could compare the current commit against the commit of the tag and change the version number if its not the same. Semantic versioning allows for additions to the version number that help explain its origin. The current RTI code, however, doesn't use semver but some binary encoding...
